common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanConfig.java

```diff
@@ -0,0 +1,387 @@
+                            api.link("Github source", "https://github.com/" + repoPath + "/tree/" + gitCommitId);
+                        }
+                    } else if (originUrl.contains("gitlab.com/") || originUrl.contains("gitlab.com:")) {
+                        Matcher matcher = Pattern.compile("(.*)gitlab\\.com[/|:](.*)").matcher(originUrl);
+                        if (matcher.matches()) {
+                            String rawRepoPath = matcher.group(2);
+                            String repoPath = rawRepoPath.endsWith(".git") ? rawRepoPath.substring(0, rawRepoPath.length() - 4) : rawRepoPath;
+                            api.link("GitLab Source", "https://gitlab.com/" + repoPath + "/-/commit/" + gitCommitId);
+                        }
+                    }
+                }
+            }
+            if (!isNullOrEmpty(gitBranchName)) {
+                api.tag(gitBranchName);
+                api.value("Git branch", gitBranchName);
+            }
+            if (!isNullOrEmpty(gitStatus)) {
+                api.tag("Dirty");
+                api.value("Git status", gitStatus);
+            }
+        });
+    }
+
+    private static boolean isCi() {
+        return isJenkins() || isTeamCity() || isCircleCI() || isBamboo() || isGitHubActions() || isGitLab() || isTravis();
+    }
+
+    private static boolean isJenkins() {
+        return envVariablePresent("JENKINS_URL");
+    }
+
+    private static boolean isTeamCity() {
+        return envVariablePresent("TEAMCITY_VERSION");
+    }
+
+    private static boolean isCircleCI() {
+        return envVariablePresent("CIRCLE_BUILD_URL");
+    }
+
+    private static boolean isBamboo() {
+        return envVariablePresent("bamboo_resultsUrl");
+    }
+
+    private static boolean isGitHubActions() {
+        return envVariablePresent("GITHUB_ACTIONS");
+    }
+
+    private static boolean isGitLab() {
+        return envVariablePresent("GITLAB_CI");
+    }
+
+    private static boolean isTravis() {
+        return envVariablePresent("TRAVIS_JOB_ID");
+    }
+
+    private static boolean isGitInstalled() {
+        Runtime runtime = Runtime.getRuntime();
+        Process process = null;
+        try {
+            process = runtime.exec(new String[]{"git", "--version"});
+            boolean finished = process.waitFor(10, TimeUnit.SECONDS);
+            return finished && process.exitValue() == 0;
+        } catch (IOException | InterruptedException ignored) {
+            return false;
+        } finally {
+            if (process != null) {
+                process.destroyForcibly();
+            }
+        }
+    }
+
+    @SuppressWarnings("UnstableApiUsage")
+    private static String execAndGetStdOut(String... args) {
+        Runtime runtime = Runtime.getRuntime();
+        Process process;
+        try {
+            process = runtime.exec(args);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
```suggestion
        } finally {
            process.destroyForcibly();
        }
```

Reply:
I don't think this will work, since the process is used in the subsequent try/finally block. But you're right, we could be more careful about cleaning up the process.